### ISSUE TITLE
[1.19][Crash]: Crash on world generation

### ISSUE THREAD

**Issue by greensad** [CorgiTaco/BYG#1101]
### BYG Version

2.0.1.0

### Terrablender Version

2.0.1.136

### Mod Loader

Quilt

### Mod Loader Version

0.18.3

### What happened?

Game crashes consistently when generating a new world. Must be some kind of incompatibility with another mod I have but did not occur on 2.0.0.13. Apologies, I do not have time to sift through my mods currently. 

### crash-log

https://gist.github.com/greensad/896270833f54edd2ce4eb9e8645b6c6d

### latest.log

https://gist.github.com/greensad/7124f9c5ec05dd3809df2462e11953b6

### Did you isolate BYG?

I tried BYG alone

**Comment by Anewri32** [comment]
I also have the same problem, apparently it is related to the mods that add biomes.

**Comment by itsdinkd** [comment]
> I also have the same problem, apparently it is related to the mods that add biomes.


They wrote that they tried BYG alone so I don't believe its just because of other mods that add biomes

**Comment by whorantheironhorse** [comment]
I came here to report that same crash! From what I can tell it's an incompatibility with Terralith, removing it lets the world generate without a crash.

**Comment by Anewri32** [comment]
> > I also have the same problem, apparently it is related to the mods that add biomes.
> 
> They wrote that they tried BYG alone so I don't believe its just because of other mods that add biomes

The error may be with the BGY alone, but in my case it crashed when I used another mod to add biomes, removing them solved the problem. It would not hurt to review this as well since this did not happen before.

**Comment by greensad** [comment]
> Game crashes consistently when generating a new world. Must be some kind of incompatibility with another mod I have

Tried with BYG alone and the error did not occur as indicated here: "Game crashes consistently when generating a new world. Must be some kind of incompatibility with another mod I have"

**Comment by catter1** [comment]
Hello, I am one of the developers of Terralith, and I was investigating this Feature Order Cycle. I believe I have discovered the reason for this lies in your **Maple Taiga** biome. Here is why:

For the feature order cycle, in general terms, it should go like this:
- various taiga features
- `minecraft:patch_dead_bush`
- `minectaft:patch_grass_savanna`

But in between BYG `2.0.0.13` and BYG `2.0.1.0`,  `minecraft:patch_grass_savanna` was added to **Maple Taiga** ([as shown here](https://github.com/AOCAWOL/BYG/compare/e7549ea..b4724e5#diff-9289b17111824dfdd1201062af4c005b6cf97bc3dbcb0f5f8c3049d3ca3cce51R1045)) in this order:
- `minecraft:patch_grass_savanna`
- various taiga features

This needs to be flipped around.

I think the reason BYG `2.0.1.0` only crashed when Terralith was added, is because Terralith's **Arid Highlands** is the only biome I could find in both BYG, Terralith, and Vanilla that contains both `minecraft:patch_dead_bush` and `minectaft:patch_grass_savanna`, causing the Feature Order Cycle error.
